### PR TITLE
Book index

### PR DESCRIPTION
**Pull request by echon006**
# Description

Please check the type of change your PR introduces:
- [x] Bugfix (non-breaking change which fixes an issue)
- [x] Feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

 
Discover page search working as intended. Book title's are links to show page. 
 
 

### Type of change

Please check the relevant option.

- [x] Non-Breaking Change
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)
- [ ] Big Breaking change (this breaks everything and I am sorry)

# How Has This Been Tested?
Fixed a couple test to pass, added some as well. 

Please check the option(s) that best describe your tests. 

- [ ] New spec files were added
- [x] Tests were added to existing spec files
- [x] Some tests were changed
- [ ] No tests were changed
- [ ] All tests were changed (please explain what in God's name happened)

### Tests Description

- Please add further details of your tests here.

# Checklist:

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have removed all unused code or commented why I left it
- [ ] I have deployed my changes to a test-heroku and they work
- [x] My changes generate no new warnings
- [x] New and existing tests pass locally with my changes

### CODE REVIEW

**Review by Malllll12 — APPROVED** [review]

File: app/poros/book.rb [hunk 1]
```diff
@@ -5,14 +5,19 @@ class Book
...
               :book_cover,
               :rating,
-              :genre
+              :genre,
+              :id
+
   def initialize(data)
+    @id = data[:id]
     @title = data[:attributes][:title]
-    @author = data[:attributes][:authors][0]
```
Comment (Sierra-T-9598) [inline app/poros/book.rb:11]: Nice, I just updated this in my PR too

File: spec/facades/book_facade_spec.rb [hunk 2]
```diff
@@ -7,8 +7,8 @@
     expect(book.count).to eq(10)
   end
 
-  xit 'returns book by id', :vcr do
+  it 'returns book by id', :vcr do
     all_books = BookFacade.book_show("ZV9DDwAAQBAJ")
```
Comment (Sierra-T-9598) [inline spec/facades/book_facade_spec.rb:11]: Might be good to call this variable `book` since we are just getting one back?

**Review by Sierra-T-9598 — CHANGES_REQUESTED** [review]
Just that one variable name change would be great just for clarity :)

File: spec/facades/book_facade_spec.rb (hunk 2 above)
Comment (echon006) [reply spec/facades/book_facade_spec.rb]: Unsure who the test but I have updated the  variable to make more sense! 
all_books is now book

**Review by devin-p-lay — APPROVED** [review]
with the before mentioned change request being resolved, this looks good

**Review by Sierra-T-9598 — APPROVED** [review]

**Review by jbreit88 — APPROVED** [review]

**Pull request merged**